File: valoration-service/src/main/java/uhu/ulises/entity/Valoracion.java
package uhu.ulises.entity;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;



@Entity
@Table(name = "valoraciones")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Valoracion {
	
	@EmbeddedId
	private ValoracionIdentity valoracionIdentity;
	
	@Min(1)
	@Max(5)
	private int puntuacion;
	
	@Size(min = 0, max = 511)
	private String comentario;
	
	@Column(name = "fecha_valoracion")
	@NotNull
	private Date fechaValoracion;
}

File: AplicacionWebTFG/AplicacionWebTFG/Models/Valoracion.cs
namespace AplicacionWebTFG.Models
{
    public class Valoracion
    {
        public string comentario { get; set; }
        public float puntuacion { get; set; }
        public ValoracionIdentity valoracionIdentity { get; set; }
        public DateTime fechaValoracion;
        public string tituloPublicacion { get; set; }
        public string nombre;
        
        public long idPublicacion
        {
            set
            {
                if (this.valoracionIdentity == null)
                    this.valoracionIdentity = new ValoracionIdentity();

                this.valoracionIdentity.idPublicacion = value;
            }
            get
            {
                if (this.valoracionIdentity == null)
                    return 0;

                return this.valoracionIdentity.idPublicacion;
            }
        }

        public string usuarioConsumidor
        {
            set
            {
                if (this.valoracionIdentity == null)
                    this.valoracionIdentity = new ValoracionIdentity();

                this.valoracionIdentity.usuarioConsumidor = value;
            }
            get
            {
                if (this.valoracionIdentity == null)
                    return null;

                return this.valoracionIdentity.usuarioConsumidor;
            }
        }
    }
}
